### PR TITLE
Fix bug for CreatedOnUtc DateTimeOffset in TelemetryMessage

### PR DESCRIPTION
**Pull request by abhipsaMisra**
Fix for #3115 : Invalid cast on CreationTimeUtc

### CODE REVIEW

**Review by schoims — APPROVED** [review]

**Review by brycewang-microsoft — APPROVED** [review]

**Review by tmahmood-microsoft — APPROVED** [review]

File: e2e/test/helpers/RetryOperationHelper.cs [hunk 1]
```diff
@@ -78,7 +78,7 @@ public class RetryOperationHelper
                     await asyncOperation().ConfigureAwait(false);
                     return;
                 }
-                catch (Exception ex) when (!retryPolicy.ShouldRetry(counter, ex, out retryInterval))
+                catch (Exception ex) when (retryPolicy.ShouldRetry(counter, ex, out retryInterval))
```
Comment (drwill-ms) [inline e2e/test/helpers/RetryOperationHelper.cs:81]: This doesn't seem right to me.

If we should retry, then we should not let the exception be thrown.

**Pull request merged**